clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/CloudMetric.groovy

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.model
+
+/**
+ * A marker interface to describe a metric reported by a cloud provider.
+ *
+ * Implementations should add any fields necessary to uniquely identify a particular metric; for example, AWS
+ * supplies "metricName" and "namespace" fields, as well as a collection of "dimensions"
+ */
+interface CloudMetric {
+
```

Comment:
This interface is consumed by the controller via the `getMetric` and `listMetrics` methods. The primary purpose of those endpoints (for now, at least) is to supply data to help the user set up autoscaling policies in the UI.

You're right - metrics should at least have a `name`. I will add that to the interface. I also agree that `type`/`unit` would be nice. For whatever reason, AWS doesn't supply the unit of measurement except in the context of an aggregated data point, so determining what the unit should be is non-trivial - you'd have to get data from a time window, and it's possible there's no data, in which case you can't determine it at all. So I'm leaving it off.

To answer your last question - units are all three of those things. These are the possible values AWS will send in the response to getStatistics:

> Seconds, Microseconds, Milliseconds, Bytes, Kilobytes, Megabytes, Gigabytes, Terabytes, Bits, Kilobits, Megabits, Gigabits, Terabits, Percent, Count, Bytes/Second, Kilobytes/Second, Megabytes/Second, Gigabytes/Second, Terabytes/Second, Bits/Second, Kilobits/Second, Megabits/Second, Gigabits/Second, Terabits/Second, Count/Second, None